ramp-frontend/ramp_frontend/views/ramp.py

```diff
@@ -80,10 +80,37 @@ def problems():
         add_user_interaction(
             db.session, interaction='looking at problems', user=user
         )
+    problems = get_problem(db.session, None)
+
+    for problem in problems:
+        for event in problem.events:
+            # check the state of the event
+            now = datetime.datetime.now()
+            start = event.opening_timestamp
+            start_collab = event.public_opening_timestamp
+            end = event.closing_timestamp
+            if now < start or now >= end:
+                event.state = 'close'
```

Comment:
So this is setting an attribute to a SQLAlchemy table object for a field that doesn't exist in the database (and so it has no impact on the database), right? It might be worth mentioning it in a comment.